### PR TITLE
build: Fix clippy and make CI respect top-level Cargo.toml

### PR DESCRIPTION
**Pull request by def-**
Fails clippy: https://buildkite.com/materialize/test/builds/85270#019090cf-d413-40fa-a0df-26da0e35e7dd

Started with https://github.com/MaterializeInc/materialize/pull/28057

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/repr/benches/packed.rs [hunk 1]
```diff
@@ -29,7 +29,7 @@ fn bench_interval(c: &mut Criterion) {
 
     const INTERVAL: Interval = Interval::new(1, 1, 0);
     group.bench_function("encode", |b| {
-        let mut buf = vec![0u8; 32];
+        let mut buf = [0u8; 32];
```
Comment (nrainer-materialize) [inline src/repr/benches/packed.rs:32]: Are these changes intended?

Comment (nrainer-materialize) [reply src/repr/benches/packed.rs:32]: Ah, I see, there were warnings.

**Review by nrainer-materialize — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
The linter is failing.

**Review by nrainer-materialize — APPROVED** [review]

**Comment by doy-materialize** [comment]
@def-: why did we change the mod_module_files lint here?

**Comment by def-** [comment]
The mz-interchange benchmark generates a mod module file. It was not caught previously since we didn't run all targets when linting.

**Comment by doy-materialize** [comment]
if it's just code we are generating, could we make it autogenerate a file with a different name instead?
